src/unk_020215A0.c

```diff
@@ -0,0 +1,864 @@
+            break;
+        }
+    }
+    if (i < sObjCharTransferTasksManager->max) {
+        sub_020223C4(&sObjCharTransferTasksManager->tasks[i]);
+        if (sObjCharTransferTasksManager->tasks[i].unk_3C == 3) {
+            sObjCharTransferTasksManager->tasks[i].unk_3C = 2;
+        } else {
+            sObjCharTransferTasksManager->tasks[i].unk_3C = 0;
+            ObjCharTransferTask_Reset(&sObjCharTransferTasksManager->tasks[i]);
+        }
+    }
+}
+
+BOOL sub_02021AC8(u32 size, BOOL a1, NNS_G2D_VRAM_TYPE vram, UnkStruct_02021AC8 *a3) {
+    u32 offsetMain;
+    u32 offsetSub;
+    u32 sizeMain;
+    u32 sizeSub;
+    BOOL ret;
+
+    if (a1 == 0) {
+        ret = ObjCharTransfer_TryGetDestVramOffsets(size, vram, &offsetMain, &offsetSub);
+        if (ret) {
+            ObjCharTransfer_ReserveVramSpace(size, vram);
+            a3->vram = vram;
+            a3->size = size;
+            a3->offset = vram == NNS_G2D_VRAM_TYPE_2DMAIN ? offsetMain : offsetSub;
+            a3->unk_0A = 0;
+        }
+    } else {
+        ret = sub_02021DE4(vram, &offsetMain, &offsetSub, size, &sizeMain, &sizeSub);
+        if (ret) {
+            sub_02021EC0(vram, offsetMain, offsetSub, sizeMain, sizeSub);
+            a3->vram = vram;
+            if (vram == NNS_G2D_VRAM_TYPE_2DMAIN) {
+                a3->size = sizeMain;
+                a3->offset = offsetMain + sObjCharTransferTasksManager->unk_1C;
+            } else {
+                a3->size = sizeSub;
+                a3->offset = offsetSub + sObjCharTransferTasksManager->unk_20;
+            }
+            a3->unk_0A = 1;
+        }
+    }
+    return ret;
+}
+
+void sub_02021B5C(UnkStruct_02021AC8 *a0) {
+    if (a0->unk_0A != 0) {
+        if (a0->vram & 1) {
+            u32 r5 = sub_020224A8(a0->size, sObjCharTransferTasksManager->unk_2C);
+            u32 r0 = sub_020224A8(a0->offset - sObjCharTransferTasksManager->unk_1C, sObjCharTransferTasksManager->unk_2C);
+            sub_02022374(r0, r5, sObjCharTransferTasksManager->unk_34);
+        }
+        if (a0->vram & 2) {
+            u32 r5 = sub_020224A8(a0->size, sObjCharTransferTasksManager->unk_30);
+            u32 r0 = sub_020224A8(a0->offset - sObjCharTransferTasksManager->unk_20, sObjCharTransferTasksManager->unk_30);
+            sub_02022374(r0, r5, sObjCharTransferTasksManager->unk_38);
+        }
+    }
+}
+
+ObjCharTransferTasksManager *sub_02021BD0(void) {
+    GF_ASSERT(sObjCharTransferTasksManager != NULL);
+    ObjCharTransferTasksManager *ret = sObjCharTransferTasksManager;
+    sObjCharTransferTasksManager = NULL;
+    return ret;
+}
+
+void sub_02021BEC(ObjCharTransferTasksManager *a0) {
+    GF_ASSERT(sObjCharTransferTasksManager == NULL);
+    sObjCharTransferTasksManager = a0;
+}
+
+void ObjCharTransferTask_Reset(ObjCharTransferTask *a0) {
```

Comment:
go through and rename all the unown vars to ``task`` or the like for the now documented ObjCharTranserTask